### PR TITLE
NanoRT: add recipe

### PR DESCRIPTION
**Pull request by amrox**
Specify library name and version:  **nanort/0bb8ab5**

Add recipe for NanoRT

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by czoido** [comment]
Hi @amrox,
Thanks a lot for the contribution. Please, check the section in the docs about the versioning system in Conan Center when there are no releases for the upstream library: https://github.com/conan-io/conan-center-index/blob/master/docs/adding_packages/conanfile_attributes.md#conancenter-specific-releases-format

**Comment by czoido** [comment]
I'll do some code suggestions, in case you missed it we have some templates for the most typical types of packages, for this case, the header-only recipe is here: https://github.com/conan-io/conan-center-index/blob/master/docs/package_templates/header_only/all/conanfile.py

File: recipes/nanort/all/conanfile.py
Comment (czoido) [inline recipes/nanort/all/conanfile.py:19]: Please, add the layout and the package_id methods to the recipe:

```suggestion
    def layout(self):
        basic_layout(self, src_folder="src")

    def package_id(self):
        self.info.clear()
```

do not forget to import `from conan.tools.layout import basic_layout`

File: recipes/nanort/all/conanfile.py
Comment (czoido) [inline recipes/nanort/all/conanfile.py:17]:
```suggestion
```
We can remove the comments and the `exports_sources` is not needed.

File: recipes/nanort/all/conanfile.py
Comment (czoido) [inline recipes/nanort/all/conanfile.py:28]:
```suggestion
```

File: recipes/nanort/all/test_package/test_package.cpp
Comment (czoido) [inline recipes/nanort/all/test_package/test_package.cpp:5]: checking the examples: https://github.com/lighttransport/nanort/blob/release/examples/bidir_path_tracer/main.cc I can see that the typical include is directly like: `#include "nanort.h"`

File: recipes/nanort/all/conanfile.py
Comment (czoido) [inline recipes/nanort/all/conanfile.py:18]:
```suggestion
    no_copy_source = True
    package_type = "header-library"    
```

**Comment by czoido** [comment]
@amrox I summarized all the changes in a PR to your repo, as I was not able to push directly to your branch: https://github.com/exyntech/conan-center-index/pull/1/files

**Comment by amrox** [comment]
> @amrox I summarized all the changes in a PR to your repo, as I was not able to push directly to your branch: https://github.com/exyntech/conan-center-index/pull/1/files

Thank you! Your comments were very instructive. 

I did miss the documentation in this repo, and instead relied on https://docs.conan.io/2/. This may be why I missed some of the details for header-only libraries and projects without versions.

**Comment by czoido** [comment]
> > @amrox I summarized all the changes in a PR to your repo, as I was not able to push directly to your branch: https://github.com/exyntech/conan-center-index/pull/1/files
> 
> Thank you! Your comments were very instructive.
> 
> I did miss the documentation in this repo, and instead relied on https://docs.conan.io/2/. This may be why I missed some of the details for header-only libraries and projects without versions.

Thanks to you for the contribution, I think this is almost ready, let's wait for CI, after that it will need a second approval from other reviewer but it should be merged very soon.

File: recipes/nanort/all/conanfile.py
Comment (czoido) [inline recipes/nanort/all/conanfile.py:30]:
```suggestion
        copy(self, pattern="LICENSE", dst=os.path.join(self.package_folder, "licenses"), src=self.source_folder)
        copy(self, "nanort.h", self.source_folder, os.path.join(self.package_folder, "include"))
```

File: recipes/nanort/all/conanfile.py
Comment (czoido) [inline recipes/nanort/all/conanfile.py:4]:
```suggestion
from conan.tools.build import check_min_cppstd

```

File: recipes/nanort/all/conanfile.py
Comment (czoido) [inline recipes/nanort/all/conanfile.py:19]:
```suggestion

        @property
    def _min_cppstd(self):
        return 11

    def validate(self):
        if self.settings.compiler.get_safe("cppstd"):
            check_min_cppstd(self, self._min_cppstd)

```

The project needs C++11 so we have to validate that we are using the correct configuration

File: recipes/nanort/all/test_package/CMakeLists.txt
Comment (czoido) [inline recipes/nanort/all/test_package/CMakeLists.txt:8]:
```suggestion
target_link_libraries(${PROJECT_NAME} PRIVATE nanort::nanort)
target_compile_features(${PROJECT_NAME} PRIVATE cxx_std_11)

```
This is needed for our v1 pipeline, because we don't pass cppstd to the build, but in v2 would not be necessary as it's already set by Conan coming from the profile.

**Comment by czoido** [comment]
@amrox I added a couple more of suggestions related to the cppstd, please check them

File: recipes/nanort/all/conanfile.py
Comment (czoido) [inline recipes/nanort/all/conanfile.py:21]:
```suggestion
    @property
```

**Review by czoido — APPROVED** [review]

**Review by davidsanfal — APPROVED** [review]

**Pull request merged**
